src/txn/txn_rollback_to_stable.c

```diff
@@ -265,6 +239,323 @@ __rollback_check_if_txnid_non_committed(WT_SESSION_IMPL *session, uint64_t txnid
     return (found);
 }
 
+/*
+ * __rollback_col_ondisk_fixup_key --
+ *     Allocate tombstone and calls function add update to head of insert list.
+ */
+static int
+__rollback_col_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_COL *cip,
+  wt_timestamp_t rollback_timestamp, bool replace, uint64_t recno)
+{
+    WT_CELL *kcell;
+    WT_CELL_UNPACK_KV *unpack, _unpack;
+    WT_CURSOR *hs_cursor;
+    WT_CURSOR_BTREE *cbt;
+    WT_DECL_ITEM(hs_key);
+    WT_DECL_ITEM(hs_value);
+    WT_DECL_ITEM(key);
+    WT_DECL_RET;
+    WT_ITEM full_value;
+    WT_PAGE *page;
+    WT_UPDATE *hs_upd, *tombstone, *upd;
+    wt_timestamp_t hs_durable_ts, hs_start_ts, hs_stop_durable_ts, newer_hs_durable_ts;
+    uint64_t hs_counter, type_full;
+    uint32_t hs_btree_id;
+    uint8_t *p;
+    uint8_t type;
+    int cmp;
+    char ts_string[5][WT_TS_INT_STRING_SIZE];
+    bool valid_update_found;
+#ifdef HAVE_DIAGNOSTIC
+    bool first_record;
+#endif
+
+    page = ref->page;
+    hs_upd = upd = tombstone = NULL;
+    hs_cursor = NULL;
+    hs_durable_ts = hs_start_ts = hs_stop_durable_ts = WT_TS_NONE;
+    hs_btree_id = S2BT(session)->id;
+    WT_CLEAR(full_value);
+    valid_update_found = false;
+#ifdef HAVE_DIAGNOSTIC
+    first_record = true;
+#endif
+
+    /* Allocate buffers for the data store and history store key. */
+    WT_ERR(__wt_scr_alloc(session, WT_INTPACK64_MAXSIZE, &key));
+    WT_ERR(__wt_scr_alloc(session, 0, &hs_key));
+    WT_ERR(__wt_scr_alloc(session, 0, &hs_value));
+
+    /* Get the full update value from the data store. */
+    unpack = &_unpack;
+    kcell = WT_COL_PTR(page, cip);
+    __wt_cell_unpack_kv(session, page->dsk, kcell, unpack);
+    WT_ERR(__wt_page_cell_data_ref(session, page, unpack, &full_value));
+    WT_ERR(__wt_buf_set(session, &full_value, full_value.data, full_value.size));
+    newer_hs_durable_ts = unpack->tw.durable_start_ts;
+    p = key->mem;
+    WT_ERR(__wt_vpack_uint(&p, 0, recno));
+    key->size = WT_PTRDIFF(p, key->data);
+
+    /* Open a history store table cursor. */
+    WT_ERR(__wt_hs_cursor_open(session));
+    hs_cursor = session->hs_cursor;
+    cbt = (WT_CURSOR_BTREE *)hs_cursor;
```

Comment:
There may be a macro for this WT_HS_CBT?

Reply:
It looks like cbt = __wt_curhs_get_cbt(hs_cursor) is used in the history store restructure PR, I can make this change with the rest of the changes required with the merging of that PR.

Reply:
I changed __rollback_row_ondisk_fixup_key to __rollback_ondisk_fixup_key and it now handles both row and col scenarios, so this is fixed with the changes from the most recent develop.